numpy/_core/src/multiarray/array_method.c

```diff
@@ -261,6 +255,16 @@ fill_arraymethod_from_slots(
      */
     for (PyType_Slot *slot = &spec->slots[0]; slot->slot != 0; slot++) {
         switch (slot->slot) {
+            case NPY_METH_resolve_descriptors_with_scalars:
+                if (!private) {
+                    PyErr_SetString(PyExc_RuntimeError,
+                            "the NPY_METH_resolve_descriptors_with_scalars "
+                            "slot private due to uncertainty about the right "
+                            "signature (see gh-24915)");
+                    return -1;
+                }
```

Comment:
I'm fine with coming back to this when someone has a real need for it.

Reply:
Same here, fine to come back to this.

Reply:
Not adding, I can't see a comment that does more than repeat the error message.  Added a leading underscore to the slot ID, though, so it is less awkward to keep it in a public header.